### ISSUE TITLE
Codename bullseye -> bookworm

### ISSUE THREAD

**Issue by EysteinWaade** [Azure/azure-functions-docker#905]
Hi,

I am running into problems running apt-get commands in docker files using the following image:
mcr.microsoft.com/azure-functions/python:4-python3.10

E: Repository 'http://security.debian.org/debian-security stable-security InRelease' changed its 'Codename' value from 'bullseye-security' to 'bookworm-security'

**Comment by miqm** [comment]
The problem is - again - not pinning the base docker image to concrete debian version and assuming stable will always be this same version:
This line and similar should use concrete debian versions like bullseye:
https://github.com/Azure/azure-functions-docker/blob/dev/host/4/bullseye/amd64/python/python310/python310-slim.Dockerfile#L66

plus - and this is the most important part:
the line with `FROM` https://github.com/Azure/azure-functions-docker/blob/dev/host/4/bullseye/amd64/python/python310/python310-slim.Dockerfile#L43 
should have the debian version specified.

Alternatively, use a script to extract current debian version when adding sources.list

**Comment by miqm** [comment]
@pragnagopa @CooperLink @gavin-aguiar @vrdmr - can you take a look?

**Comment by EysteinWaade** [comment]
Our quick fix to the problem was to downgrade to image 4.17.3

**Comment by JanDubcak** [comment]
As a temp workaround I have switched to nightly build of docker image python3.10.

**Comment by BrittaMeixner** [comment]
We have the same issue, no matter where we build (DevOps/locally)
#0 0.672 Get:1 http://security.debian.org/debian-security stable-security InRelease [48.0 kB]
#0 0.675 Get:2 http://deb.debian.org/debian bullseye InRelease [116 kB]
#0 0.714 Get:3 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
#0 0.736 Get:4 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
#0 0.842 Get:5 https://packages.microsoft.com/debian/11/prod bullseye InRelease [3629 B]
#0 0.914 Get:6 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [245 kB]
#0 1.062 Get:7 https://packages.microsoft.com/debian/11/prod bullseye/main armhf Packages [13.9 kB]
#0 1.087 Get:8 https://packages.microsoft.com/debian/11/prod bullseye/main amd64 Packages [87.1 kB]
#0 1.135 Get:9 https://packages.microsoft.com/debian/11/prod bullseye/main arm64 Packages [15.3 kB]
#0 1.146 Reading package lists...
#0 1.611 E: Repository 'http://security.debian.org/debian-security stable-security InRelease' changed its 'Codename' value from 'bullseye-security' to 'bookworm-security'
 
Assuming it is around line 71 of https://github.com/Azure/azure-functions-docker/blob/dev/host/4/bullseye/amd64/python/python310/python310.Dockerfile

Related to changes in Debian over the weekend. Nothing in our build nor dockerfile changed. Just started failing this morning.

**Comment by CooperLink** [comment]
Thank you for filing this issue and for the detailed notes about how it is affecting you.  I was able to reproduce your issue with the image you mentioned : mcr.microsoft.com/azure-functions/python:4-python3.10.

This issue has started to occur due to the [Debian 12 release](https://www.debian.org/News/2023/20230610) that took place on June 10th 2023.  As a part of this release the stable-security source list for Debian images was changed to a Debian 12 source list.  The error that is being thrown does not appear to be fatal on my reproduction.  The cause of the error is that our images were built with a different stable-security target than what is currently available.  The issue is not occurring in images built -nightly as @JanDubcak mentioned above. 

The error appears to be non-fatal to me.  We will still work to deliver a mitigation swiftly, but could anyone here help me understand how this is blocking your workflows?  I see the error log printed above in my local repro but, apt-get update/install still succeeds.  I was able to install additional libraries as well.   

My current belief is that this is only blocking builds that use one of our python images as a base image.  A quick mitigation for this would be to move to the -nightly image tag. We will update this issue when we are able to update major tags. 

In this [PR](https://github.com/Azure/azure-functions-docker/pull/908), we will address this issue in Python Docker Images by updating the location added to the source list as discussed above.  This update will be deployed to the image major version tags within the week.  In the meantime, please consider using the 4-nightly tag as a workaround for your builds

Ex. mcr.microsoft.com/azure-functions/python:4-nightly-python3.10

**Comment by miqm** [comment]
@CooperLink I commented on the PR regarding the base python image you use. Year or two ago there was a similar issue when python started using bullseye instead buster. I assume this will happen soon, so I'd recommend to sticking with concrete debian version and upgrading it consciously rather than relying on moving stable image tag.

**Comment by CooperLink** [comment]
@miqm Thank you for the recommendation.  I am discussing doing this with the Python language team. If a debian 11/ bullseye tag is currently available we will swap to it and if not we will start the process to request one.

**Comment by miqm** [comment]
Of course it is. Just add -bullseye suffix to image tag (i.e. `4-slim-bullseye`). It'd be awesome if azure functions host would also give this ability (choosing debian version by selecting a base tag) in future.

**Comment by CooperLink** [comment]
Images have been refreshed with updated dependencies.  With the new images I am no longer getting the error logs indicated above. Please let me know if you are still blocked on this at this time.

**Comment by BrittaMeixner** [comment]
For us it is all working again. Locally and in DevOps.

**Comment by CooperLink** [comment]
Great! I will keep this issue open until the end of the day today to see if anyone else is blocked/ needs assistance.

**Comment by miqm** [comment]
Under which tag/version is the fix released? I see that mentioned PR is still open and there's no new GH release on this repo.

**Comment by CooperLink** [comment]
The updated build is available on our the following two tags :4.21.1-refresh and our floating tag :4 major version.   The PR will be closed and released in the next available Azure Functions Release.

ex. mcr.microsoft.com/azure-functions/python:4-python3.10

**Comment by vivekjilla** [comment]
Thanks for the report. This was fixed by refreshed images that corrected the apt sources, and multiple users confirmed it working again. Closing as resolved — if you run into apt/package-source errors on a current supported image tag/digest, please open a new issue and we'll take a look.
